### ISSUE TITLE
mix() helper does not respect baseUrl

### ISSUE THREAD

**Issue by pedzed** [tightenco/jigsaw#254]
Perhaps I misunderstood something, but the default codebase comes with the `mix()` helper in the layout. The default codebase also comes with `baseUrl` in the config, but I am not sure what `baseUrl` is for, if it is purposely not used in the `mix()` helper.

https://github.com/tightenco/jigsaw/blob/983df583f31ed02f2a7484a0e663ab5fedd7cb4e/stubs/site/config.php#L4

https://github.com/tightenco/jigsaw/blob/983df583f31ed02f2a7484a0e663ab5fedd7cb4e/stubs/site/source/_layouts/master.blade.php#L7

**Comment by raniesantos** [comment]
I'm not very familiar with Jigsaw's mix setup either. I have problems making it compile fonts with the defaults, I had to change the directory structure of the css to make it work properly.

**Comment by damiani** [comment]
The `baseUrl` setting in `config.php` has nothing to do with asset compilation; it's used for generating an absolute URL using the `$page->getUrl()` command, as opposed to `$page->getPath()` which generates a path relative to your site root. 

To control the destination of assets being compiled by Laravel Mix, you would modify the Webpack configuration in `webpack.mix.js`: `mix.setPublicPath("source/assets/build");` is there by default, and corresponds to the default setup in `master.blade.php`.

**Comment by pedzed** [comment]
@damiani Thanks for clarifying. I realized it after posting.

So how can I do what I want? Should I just ignore the `mix()` helper and write my own function to check for `baseUrl`?

**Comment by damiani** [comment]
The `mix()` helper generates a relative URL, so I would just prepend your `baseUrl` to it:

```
<link rel="stylesheet" type="text/css" href="{{ trim($page->baseUrl, '/') }}{{ mix('css/style.css', 'assets/build') }}">
```

Then set different `baseUrl`s in `config.php` and `config.production.php`:

> config.php
```
'baseUrl' => 'http://localhost:3000',
```

> config.production.php
```
'baseUrl' => 'https://subdomain.example.com/subfolder',
```

**Comment by pedzed** [comment]
That's really ugly. But thanks, I think I know enough to write my own `asset()` helper function now.

---- On Thu, 23 Aug 2018 23:55:51 +0200 notifications@github.com wrote ----

The mix() helper generates a relative URL, so I would just prepend your baseUrl to it:

<link rel="stylesheet" type="text/css" href="{{ trim($page->baseUrl, '/') }}{{ mix('css/style.css', 'assets/build') }}">
Then set different baseUrls in config.php and config.production.php:

config.php

'baseUrl' => 'http://localhost:3000',
config.production.php

'baseUrl' => 'https://subdomain.example.com/subfolder',

**Comment by pedzed** [comment]
Just one more request. Could the docs be expanded a bit more? Maybe I will contribute with some PR's, but it would be nice if the default codebase was clearer on how things work.

---- On Fri, 24 Aug 2018 00:01:59 +0200 notifications@github.com wrote ----

Closed #254.

**Comment by damiani** [comment]
PRs to docs are always welcome!
